### PR TITLE
Task 7

### PR DESCRIPTION
**Pull request by MJahidH**

### CODE REVIEW

File: src/utils.js [hunk 1]
```diff
@@ -1,9 +1,26 @@
...
+    });
+};
 
-return axios.get("https://nc-news-068s.onrender.com/api/articles").then((res)=>{
-    return res.data.articles
+
+export const patchVotes = (inc_votes,article_id) => {
+
+    return axios.patch(`https://nc-news-068s.onrender.com/api/articles/${article_id}`,inc_votes).then((res) => {
+    console.log(res.data,"utiles")
```
Comment (nc-modou) [inline src/utils.js]: Remember to remove logs

File: src/components /MainArticleCard.jsx [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+
+  const fetchArticle = () => {
+    getArticleById(article_id).then((article) => {
+      setArticle(article);
+    });
+  };
+
+  useEffect(() => {
+    fetchArticle();
+  }, []);
```
Comment (nc-modou) [inline src/components /MainArticleCard.jsx]: You're not doing anything inside this useEffect at the moment... It is making a request with your fetchArticle function but it's not doing anything with the result of it. I don't think this is necessary from the looks of things

File: src/components /MainArticleCard.jsx [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+
+  const handleClick = (event) => {
+    patchVotes({ inc_votes: 1 }, article_id);
+    fetchArticle();
```
Comment (nc-modou) [inline src/components /MainArticleCard.jsx]: I see what you were intending by fetching the article after making a vote - it definitely does work, but there's a delay between clicking upvote and seeing the votes actually go up.

And also - you don't have to make a separate fetch in the first place. Your patch request itself returns an updated article object (with the increased votes) meaning that you don't have to make a separate GET request to get the new article.

Also, we wanna make comment votes optimistic - meaning that we assume the vote is gonna work, increase the votes by 1, and only reverse that if the request fails. This notes page might be helpful!

https://notes.northcoders.com/courses/js-front-end/ui-behaviours-and-optimistic-rendering

**Review by nc-modou — COMMENTED** [review]
Great work so far Jahid. Upvoting works, we just want to make this optimistic too. Take a look at the feedback comments and feel free to send me a message if you need anything clarified

**Review by popatre — CHANGES_REQUESTED** [review]
Nice optimistic rendering - however, we need some feedback for when the request fails. At the moment, the number goes up which makes it look like it worked. 

We need to be notified the request has failed and to make sure the number doesnt stay incremented.

**Review by popatre — APPROVED** [review]
Nice error message, although it takes over the whole screen and stops me viewing the article. Ideally, there would be a little message next to the button that something gone wrong, so i can continue to view the article. 

I'll approve it for now, but maybe something to think about as a refactor later on/come back to

**Pull request merged**